src/Storages/ExternalStream/Kafka/Kafka.h

```diff
@@ -42,7 +42,8 @@ class Kafka final : public StorageExternalStreamImpl
     const String & password() const { return settings->password.value; }
     const String & sslCaCertFile() const { return settings->ssl_ca_cert_file.value; }
     const klog::KConfParams & properties() const { return kafka_properties; }
-    const ASTPtr & partitioning_expr_ast() const { return partitioning_expr; }
+    bool has_custom_partitioning_expr() const { return !engine_args.empty(); }
```

Comment:
Let's use camelCase convention

Reply:
Is partitioning expr meaning sharding expression, if yes, please use sharding expression consitently since partition in Proton database is totally different.

Reply:
```
hasShardingExpr() ...
shardingExprAst() ...
```

Reply:
I am trying to use partition/partitioning in the context of Kafka, and shard/sharding in other places. This is because all Kafka APIs use partition not shard. Do you still think we should use shard even in the context of Kafka?

Reply:
I prefer sharding everywhere.